File: src/domain/entities/transformer/Transformer.ts
import type { VoltageReading } from "../voltage/VoltageReading";
import type { TransformerHealth } from "./TransformerHealth";

export interface Transformer {
  assetId: number;
  name: string;
  region: string;
  health: TransformerHealth;
  voltageReadings: VoltageReading[];
}

File: backend/src/TransformerMonitor.Domain/Entities/Transformer.cs
namespace TransformerMonitor.Domain.Entities;

public class Transformer
{
    public int Id { get; set; }
    public int AssetId { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Region { get; set; } = string.Empty;
    public bool IsActive { get; set; } = true;
    public double BaseVoltage { get; set; } = 22000;
    public ICollection<VoltageReading> VoltageReadings { get; set; } = new List<VoltageReading>();
    public ICollection<Ticket> Tickets { get; set; } = new List<Ticket>();
}
